extensions/blocks/opentable/opentable.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * OpenTable Block.
+ *
+ * @since 8.x
+ *
+ * @package Jetpack
+ */
+
+if ( jetpack_opentable_block_is_available() ) {
+	jetpack_register_block(
+		'jetpack/opentable',
+		array( 'render_callback' => 'jetpack_opentable_block_load_assets' )
+	);
+} else {
+	Jetpack_Gutenberg::set_extension_unavailable(
+		'jetpack/opentable',
+		'missing_plan',
+		array(
+			'required_feature' => 'opentable',
+			'required_plan'    => ( defined( 'IS_WPCOM' ) && IS_WPCOM ) ? 'value_bundle' : 'jetpack_premium',
+		)
+	);
+}
+
+/**
+ * Is the OpenTable block available on a given site
+ *
+ * @return bool True if the block is available, false otherwise.
+ */
+function jetpack_opentable_block_is_available() {
+	// For WPCOM sites.
+	if ( defined( 'IS_WPCOM' ) && IS_WPCOM && function_exists( 'has_any_blog_stickers' ) ) {
+		$site_id = jetpack_get_blog_id();
+		return has_any_blog_stickers( array( 'premium-plan', 'business-plan', 'ecommerce-plan' ), $site_id );
+	}
+	// For all Jetpack sites.
+	return Jetpack::is_active() && Jetpack_Plan::supports( 'opentable' );
```

Comment:
Since the block doesn't rely on the WordPress.com API to work, I am not sure we can pay gate its access, as per rule 5 of the guidelines:

https://developer.wordpress.org/plugins/wordpress-org/detailed-plugin-guidelines/#5-trialware-is-not-permitted

I believe it should remain free.

Reply:
I think we've reached a decision on this one!